### DISCUSSION TITLE
Code Folder For Brackets () ?

### DISCUSSION THREAD

**Discussion by ysk3a** [FIameCaster/prism-code-editor#24] [Q&A] [answered]
I've noticed in the demos that the readonly editor does not have a code folding for regular brackets ` ( ` and `)`?
For example, it appear monaco editor will fold multi line brackets like:
![image](https://github.com/user-attachments/assets/c7ad7963-2a5f-4c71-a3e1-e8b32dca7989)
but the demo does not show the collapse option?:
![image](https://github.com/user-attachments/assets/e24eb11a-a12c-4a7f-88a2-f7c95fcca79c)

This is not an issue for me but i was wondering if this is the indented behavior or it is just something happening on my end when view the demos.

**Comment by jonpyt** [comment, marked as answer]
Folding for round brackets was disabled intentionally. I have though about changing this.

In the mean time, if you want folding for round brackets, you can add a `FoldingRangeProvider` for it like so:

```ts
import { readOnlyCodeFolding, FoldingRangeProvider } from "prism-code-editor/code-folding"
import "prism-code-editor/code-folding.css"

const isMultiline = (str: string, start: number, end: number) => str.slice(start, end).includes("\n")

const roundBracketFolding: FoldingRangeProvider = editor => {
  const matchBrackets = editor.extensions.matchBrackets
  const folds: [number, number][] = []
  const value = editor.value

  if (matchBrackets) {
    const { brackets, pairs } = matchBrackets
    for (let i = 0, , l = pairs.length; i < l; i++) {
      const other = pairs[i]!
      if (
        other > i &&
        brackets[i][3] == "(" &&
        isMultiline(value, brackets[i][1], brackets[other][1])
      ) {
        folds.push([brackets[i][5], brackets[other][1]])
      }
    }
  }

  return folds
}

editor.addExtensions(readOnlyCodeFolding(roundBracketFolding))
```

**Reply by ysk3a** [reply]
Thanks for the quick reply!
> Folding for round brackets was disabled intentionally. I have though about changing this.

If the community wants this feature or if you have plan of adding it, i think it would be a nice to have but i don't think it is a required feature. 
it was just that i had originally had the impression that 'brackets' or 'code folding' mentioned in the comments in the index.d.ts of the folding extension meant that it also included round brackets. So thank you for the clarification and for providing a sample FoldingRangeProvider.
I'll definitely try out adding FoldingRangeProvider later.
